main.tf

```diff
@@ -1,3 +1,95 @@
+  validate_cleanup_lite = [for cp0 in var.cleanup_policy : (var.plan == "lite" && (cp0 == "compact" || cp0 == "compact,delete")) ? tobool("In lite plan the only allowed cleanup policy is 'delete'") : true]
+  # tflint-ignore: terraform_unused_declarations
+  validate_number_of_topic_lite = (var.plan == "lite" && length(var.topic_names) > 1) ? tobool("Only one topic is allowed in lite plan") : true
+  # tflint-ignore: terraform_unused_declarations
+  validate_partition_lite_inputs = [for p1 in var.partitions : (var.plan == "lite" && p1 > 1 ? tobool("For lite plan partition can not exceed 1") : true)]
+  # tflint-ignore: terraform_unused_declarations
+  validate_partition_standard_inputs = [for p2 in var.partitions : (var.plan == "standard" && p2 > 100 ? tobool("For standard plan partition can not exceed 100") : true)]
+  # tflint-ignore: terraform_unused_declarations
+  validate_segment_bytes_lite_inputs = [for s0 in var.segment_bytes : (var.plan == "lite" && (s0 < 10485760 || s0 > 536870912) ? tobool("For lite plan the allowed value is [10 MiB, 500 MiB]") : true)]
+  # tflint-ignore: terraform_unused_declarations
+  validate_segment_bytes_standard_inputs = [for s1 in var.segment_bytes : (var.plan == "standard" && (s1 < 102400 || s1 > 536870912) ? tobool("For standard plan the allowed value is [100 KiB, 500 MiB]") : true)]
+  # tflint-ignore: terraform_unused_declarations
+  validate_segment_bytes_enterprise_inputs = [for s2 in var.segment_bytes : (var.plan == "enterprise-3nodes-2tb" && (s2 < 102400 || s2 > 2199023255552) ? tobool("For enterprise plan the allowed value is [100 KiB, 2 TiB]") : true)]
+  # tflint-ignore: terraform_unused_declarations
+  validate_retention_bytes_lite_inputs = [for r0 in var.retention_bytes : (var.plan == "lite" && (r0 < 10485760 || r0 > 104857600) ? tobool("For lite plan the allowed value is [10 MiB, 100 MiB]") : true)]
+  # tflint-ignore: terraform_unused_declarations
+  validate_retention_bytes_standard_inputs = [for r1 in var.retention_bytes : (var.plan == "standard" && (r1 < 102400 || r1 > 1074790400) ? tobool("For standard plan the allowed value is [100 KiB, 1 GiB]") : true)]
+  # tflint-ignore: terraform_unused_declarations
+  validate_retention_bytes_enterprise_inputs = [for r2 in var.retention_bytes : (var.plan == "enterprise-3nodes-2tb" && (r2 < 102400 || r2 > 2199023255552) ? tobool("For enterprise plan the allowed value is [100 KiB, 2 TiB]") : true)]
+  # tflint-ignore: terraform_unused_declarations
+  validate_retention_ms_lite = [for rm0 in var.retention_ms : (var.plan == "lite" && (rm0 < 3600000 || rm0 > 2592000000)) ? tobool("For lite plan allowed range for retention time is [1 hour, 30 days]") : true]
+  # tflint-ignore: terraform_unused_declarations
+  validate_retention_ms_standard = [for rm1 in var.retention_ms : (var.plan == "standard" && (rm1 < 3600000 || rm1 > 2592000000)) ? tobool("For standard plan allowed range for retention time is [1 hour, 30 days]") : true]
+}
```

Comment:
some of this validation will now need to be reviewed after doing the work to move all topic attributes into its own variable